### ISSUE TITLE
@Retry annotation does not work with Micrometer

### ISSUE THREAD

**Issue by Eyesun23** [resilience4j/resilience4j#1996]
Resilience4j version:
1.7.0
Java version: 17.0.7

The Retry annotation we have does not work with Micrometer retry?

The reason is because it does not take into account the reference.

All our code has `@Retry` annotation and we decided to use [Micrometer retry](https://resilience4j.readme.io/docs/micrometer#retry-metrics), but realized the annotation does not specify the ID/Reference.

```

import io.github.resilience4j.retry.annotation.Retry
    
@Retry(name = "backend-a")
    fun retrieve(id: String): List<User> {
        println("Calling retrieveUsers2")
        throw IllegalStateException("Test $id")
    }

```

This is my code and I should be able to add resilience4j-micrometer to the annotation without needing it to bind it. It should already bind it to the retry annotation or at least the name of the retry. This functionality is missing.

**Comment by RobWin** [comment]
Sorry, I don't understand the issue. You need to provide more context. Are you using Spring Boot? Do you use our Spring Boot Starter? What is not working?

**Comment by Eyesun23** [comment]
Yes, so the `@Retry` annotation from resilience4j does not work with the current setup you have to add `MeterRegistry`.

If I use `import io.github.resilience4j.retry.annotation.Retry`, I cannot bind Retry metrics to a MeterRegistry.

`@Retry(name = "backend-a")
fun retrieve(id: String): List {
println("Calling retrieveUsers2")
throw IllegalStateException("Test $id")`

**Comment by RobWin** [comment]
If you use our Spring Boot  Starter everything is automatically wired by our auto configuration classes.

**Comment by Eyesun23** [comment]
I don't think you understand the issue or looked at the resilience4j library itself. I have looked at all your answers, you have not resolved anyone's issue, because you didn't understand their question in first place. Resilience4j needs new member. 
The annotation does not work with Micrometer using Spring Boot, and I tried it on both Spring and Kotlin Gradle

**Comment by RobWin** [comment]
Thx and bye

**Comment by Eyesun23** [comment]
This was not addressed

**Comment by RobWin** [comment]
I really need to know how you are using Resilience4j. Please share how have configred it in Maven or Gradle.
Because we have Spring Boot demo projects and a lot of tests and everything works fine with Micrometer:  https://github.com/resilience4j/resilience4j-spring-boot2-demo

**Comment by RobWin** [comment]
If you are interested. Here are to auto configuration classes for Spring Boot.

https://github.com/resilience4j/resilience4j/tree/master/resilience4j-spring-boot2/src/main/java/io/github/resilience4j/retry/autoconfigure

And here how Micrometer is configured, when it's on the classpath:

https://github.com/resilience4j/resilience4j/blob/master/resilience4j-spring-boot2/src/main/java/io/github/resilience4j/retry/autoconfigure/RetryMetricsAutoConfiguration.java

**Comment by Eyesun23** [comment]
Sure I debugged this and I actually spin-up a spring boot as well. So the issue is not the Retry, the issue is the micrometer configuring it with the **`@Retry`** annotation.


**application.yml file:**

`resilience4j:
  retry:
    instances:
      default:
        enableExponentialBackoff: true
        exponentialBackoffMultiplier: 2
        maxAttempts: 4`


**dependencies.txt** file: 
`+--- io.github.resilience4j:resilience4j-kotlin -> 1.7.0`



**gradle.kts** file: 

`dependencies {
    implementation("io.github.resilience4j:resilience4j-kotlin")
}`


**kotlin** file where I am using the _`@Retry`_ annotation; to use micrometer based on the micrometer documentation...you need to bind it...but you cannot bind it to the retry.
`
import io.github.resilience4j.retry.annotation.Retry

@Service
class OrdersApiService(
    private val ordersApi: OrdersApi
) {
@Retry(name = Constants.DEFAULT)
    suspend fun getOrders(
        userId: Int?,
    ): GetOrderResponse =
        try {
            ordersApi.coreOrdersOrderIdGet(
                userId,
            )
                .awaitSingle()
        } catch (wcre: WebClientResponseException) {
            if (wcre.statusCode.series() == HttpStatus.Series.SERVER_ERROR) throw RetryableException(wcre) else throw wcre
        }
}`

**Comment by Eyesun23** [comment]
This is from the resilience4j library `Retry.java`:


`@Retention(value = RetentionPolicy.RUNTIME)
@Target(value = {ElementType.METHOD, ElementType.TYPE})
@Documented
public @interface Retry {

    /**
     * Name of the sync retry.
     * It can be SpEL expression. If you want to use first parameter of the method as name, you can
     * express it {@code #root.args[0]}, {@code #p0} or {@code #a0}. And method name can be accessed via
     * {@code #root.methodName}
     *
     * @return the name of the sync retry.
     */
    String name();

    /**
     * fallbackMethod method name.
     *
     * @return fallbackMethod method name.
     */
    String fallbackMethod() default "";
}
`

**Comment by RobWin** [comment]
Ok, I see.
Resilience4j annotations are not fully supported in Kotlin.
Spring AOP and Kotlin suspending functions not work properly together. 
That's why we implemented `io.github.resilience4j:resilience4j-kotlin` so that you can decorate suspending functions via code, but not annotations.  Please see: https://resilience4j.readme.io/docs/getting-started-4#usage---suspending-functions

**Comment by Eyesun23** [comment]
Yes so I verified it by using `[prometheus](https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.metrics.export.prometheus)` locally and realized that it works with the suspending functions but does not work with the `@Retry` annotation. I was hoping this functionality can be added because the Retry annotation is easy to use.

**Comment by RobWin** [comment]
Unfortunately it's not that easy  :(

**Comment by Eyesun23** [comment]
Yes but it is a bug. Why add a feature that doesn't work with other functionalities.

**Comment by RobWin** [comment]
It's not a bug. Because Resilience4j is ja Java library and not a library for Kotlin.
Just because we have a Kotlin module doesn't mean we fully support Kotlin.
Resilience4j cannot support all languages which run in JVM out-of-box. We also don't support Groovy.
But you can contribute a fix to Spring Framework and Spring AOP to fully support Kotlin, if you can.
